indexer/ai/src/main/java/com/google/cloudsearch/AISkillDriver.java

```diff
@@ -0,0 +1,42 @@
+package com.google.cloudsearch;
+
+import com.google.cloudsearch.ai.AISkill;
+import com.google.cloudsearch.ai.AISkillSet;
+import com.google.cloudsearch.exceptions.InvalidConfigException;
+import com.google.common.collect.Multimap;
+import org.json.simple.JSONObject;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class AISkillDriver {
+    private AISkillSet skillSet;
+    private Boolean isInvalidConfig = false;
+    private static Logger log = Logger.getLogger(AISkillDriver.class.getName());
+
+    public AISkillDriver(JSONObject aiConfig, JSONObject schema) {
+        try{
+            this.skillSet = new AISkillSet(aiConfig, schema);
+        }
+        catch(InvalidConfigException e){
+            this.isInvalidConfig = true;
+            log.warning("AI Skill configuration is invalid. Structured data won't be populated");
+            e.printStackTrace();
+        }
+
+    }
+    public Multimap<String, Object> populateStructuredData(Multimap<String, Object> structuredData, String filepath){
```

Comment:
why we take structuredData as argument ?

Reply:
The same structured data is added to the same structured data object by all the skills, that's why I am passing it. Alternatively, each can return we need to merge the lists, I thought adding to the same list would be better.

Reply:
each skill will populate the map fields, but why AISkillDriver gets this as argument?

Reply:
I think you wanted to have a different type of driver, but according to what I have implemented it is just a wrapper over the AI Skills, so it takes the structured data object and do the necessary to populate it.